azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CURL.java

```diff
@@ -14,11 +15,25 @@
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLEncoder;
+import java.util.Set;
 import java.util.UUID;
+import java.util.concurrent.ConcurrentSkipListSet;
 
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class AADB2CURL {
 
+    /**
+     * The ${@link Set} of generated ${@link UUID} for state field.
+     */
+    @Getter
+    private static Set<String> stateIDSet = new ConcurrentSkipListSet<>();
+
+    /**
+     * The ${@link Set} of generated ${@link UUID} for nonce field.
+     */
+    @Getter
+    private static Set<String> nonceIdSet = new ConcurrentSkipListSet<>();
```

Comment:
I see memory leak here.
Is there a built-in solution to generate nonce?

Reply:
It looks not easy to handle nonce by ourselves, will investigate some existed solution about this.